include/swift/Subsystems.h

```diff
@@ -53,6 +53,12 @@ namespace swift {
   class IRGenOptions;
   class LangOptions;
   class ModuleDecl;
+  /// A opaque syntax node created by a \c SyntaxParseAction, whose contents
+  /// must be interpreted by the \c SyntaxParseAction which created it.
+  /// Requires the two low bits to be 0, so that it can be stored in an
+  /// \c llvm::PointerIntPair. This is in particular guaranteed for pointers
+  /// to C/C++ objects and for pointers that were generated by Swift and passed
+  /// to the compiler via a C API (in particular \c CLibParseActions ).
```

Comment:
This is probably true, but just in case could you add an assertion in the `RecordedOrDeferredNode` constructor?

https://github.com/apple/llvm-project/blob/28c70a0e/llvm/include/llvm/Support/PointerLikeTypeTraits.h#L73-L74
```
  /// All clients should use assertions to do a run-time check to ensure that
  /// this is actually true.
```

Reply:
`PointerIntPairInfo` already does this check. I suppose that it is considered the client of `PointerLikeTypeTraits` and we are just using `PointerIntPairInfo`.

https://github.com/apple/llvm-project/blob/apple/main/llvm/include/llvm/ADT/PointerIntPair.h#L178